internal/bminventory/inventory.go

```diff
@@ -1486,31 +1490,27 @@ func (b *bareMetalInventory) EnableHost(ctx context.Context, params installer.En
 	return installer.NewEnableHostOK().WithPayload(&host)
 }
 
-func (b *bareMetalInventory) DownloadClusterFiles(ctx context.Context, params installer.DownloadClusterFilesParams) middleware.Responder {
+func (b *bareMetalInventory) GetPresignedForClusterFiles(ctx context.Context, params installer.GetPresignedForClusterFilesParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
-	var cluster common.Cluster
-	log.Infof("Download cluster files: %s for cluster %s", params.FileName, params.ClusterID)
-
-	if !funk.Contains(clusterFileNames, params.FileName) {
-		err := fmt.Errorf("invalid cluster file %s", params.FileName)
-		log.WithError(err).Errorf("failed download file: %s from cluster: %s", params.FileName, params.ClusterID)
-		return common.NewApiError(http.StatusBadRequest, err)
+	// Presigned URL only works with AWS S3 because scality is not exposed, in this case URL should be like s3.us-east-1.amazonaws.com
+	if !strings.HasSuffix(b.Config.S3EndpointURL, ".amazonaws.com") {
```

Comment:
maybe should be a function of the client, you have this code twice? (also empty string is also AWS)

Reply:
Good idea